src/browser-attributes-span-processor.ts

```diff
@@ -7,7 +7,7 @@ import { SpanProcessor } from '@opentelemetry/sdk-trace-base';
  * Static attributes (e.g. User Agent) are added to the Resource.
  */
 export class BrowserAttributesSpanProcessor implements SpanProcessor {
-  constructor() {}
+  constructor(public getRoute = () => window.location.pathname) {}
```

Comment:
Can we accept an options object instead of a single parameter? I don't want us to end up in a place 2 years in the future where folks are passing in 20 distinct args and the order is significant and there's no obvious way to tell which arg is which.

```ts
new BrowserAttributesSpanProcessor(
  () => { ... },
  someOtherArg,
  aThirdArg,
  true,
  false,
  yetAnotherCallback,
  ...
);
```

Reply:
for sure, I assumed the span processor was internal (and therefore easily malleable) so didn't give it much thought

Reply:
should be good now